<?xml version="1.0"?>
<robot name="mattro_rovo_2">
  
  <material name="xsens_orange">
    <color rgba="0.92 0.45 0.02 1.0"/>
  </material>

  <material name="mattro_blue">
    <color rgba="0.0 0.0 0.8 1"/>
  </material>

  <material name="plastic_white">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>

  <link name="mattro_base_link">
    <visual>
      <origin rpy="0 0 1.57079632679" xyz="0.089 0 -0.343"/>
      <geometry>
        <mesh filename="package://bock_controller/urdf/Mattro ROVO 2 - Heavy Duty - simplified.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="mattro_blue">
        <color rgba="0.2 0.2 0.8 1.0"/>
      </material>
    </visual>
  </link>

  <link name="imu_link">
    <visual>
      <origin rpy="-1.57079632679 0.0 1.57079632679" xyz="0.0 0.0 0.0016"/>
      <geometry>
        <mesh filename="package://bock_controller/urdf/MTi_6xx.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="xsens_orange"/>
    </visual>
  </link>

  <link name="control_box_link">
    <visual>
      <origin rpy="-1.57079632679 0.0 1.5707963267" xyz="0.0 0.01 0.0"/>
      <geometry>
        <mesh filename="package://bock_controller/urdf/Box_400x500x200.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="plastic_white"/>
    </visual>
  </link>

  <joint name="base_to_box" type="fixed">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    <parent link="control_box_link"/>
    <child link="mattro_base_link"/>
  </joint>

  <joint name="imu_to_box" type="fixed">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    <parent link="imu_link"/>
    <child link="control_box_link"/>
  </joint>


</robot>
